ddtrace/settings/config.py

```diff
@@ -544,5 +623,35 @@ def _get_service(self, default=None):
 
     def __repr__(self):
         cls = self.__class__
-        integrations = ", ".join(self._config.keys())
+        integrations = ", ".join(self._integration_config.keys())
         return "{}.{}({})".format(cls.__module__, cls.__name__, integrations)
+
+    def _subscribe(self, items, handler):
+        # type: (List[str], Callable[[Config, List[str]], None]) -> None
+        self._subscriptions.append((items, handler))
+
+    def _notify_subscribers(self, changed_items):
+        # type: (List[str]) -> None
+        for sub_items, sub_handler in self._subscriptions:
+            sub_updated_items = [i for i in changed_items if i in sub_items]
+            if sub_updated_items:
+                sub_handler(self, sub_updated_items)
+
+    def __setattr__(self, key, value):
+        # type: (str, Any) -> None
+        if key == "_config":
+            return super(self.__class__, self).__setattr__(key, value)
+        elif key in self._config:
+            self._config[key].set_code(value)
+            self._notify_subscribers([key])
```

Comment:
should we add `value` to the `_notify_subscribers()` call so that we can also keep track of the values being changed?

Reply:
The entire configuration is passed to subscribers so they can access the value that way. I'm not 100% about what is the better API at this point but it's easy enough to change once we start using it.